### PR TITLE
Replace laravel-mix with vite.js

### PR DESCRIPTION
**Pull request by vinkla**
This is work in progress to replace Laravel Mix with Vite.js.

### CODE REVIEW

**Review by puredazzle — CHANGES_REQUESTED** [review]
We should use `defineConfig` in Vite to get autocomplete for the config.

```
import { defineConfig } from 'vite';

export default defineConfig(({ mode }) => ({
```

**Review by puredazzle — APPROVED** [review]
LGTM! Good job!

**Comment by fiskhandlarn** [comment]
LGTM!

File: README.md [hunk 1]
```diff
@@ -159,22 +159,20 @@ The plugin should now be installed in the `public/mu-plugins` directory.
...
+[To get started with Vite, please visit the documentation.](https://vitejs.dev/guide/)
 
 ```sh
-// Run all mix tasks...
+// Start the dev server...
 npm run dev
 
-// Run all mix tasks and minify output...
+// Build for production...
 npm run build
```
Comment (fiskhandlarn) [inline README.md:173]: `vite build`

File: README.md [hunk 2]
```diff
@@ -159,22 +159,20 @@ The plugin should now be installed in the `public/mu-plugins` directory.
...
-[Laravel Mix](https://github.com/JeffreyWay/laravel-mix/tree/master/docs#readme) is a clean layer on top of Webpack to make the 80% use case laughably simple to execute. Most would agree that, though incredibly powerful, Webpack ships with a steep learning curve. But what if you didn't have to worry about that?
+[Vite](https://vitejs.dev/) is a build tool that aims to provide a faster and leaner development experience for modern web projects. Vite is opinionated and comes with sensible defaults out of the box, but is also highly extensible via its Plugin API and JavaScript API with full typing support.
 
-[To get started with Laravel Mix, please visit the documentation.](https://github.com/JeffreyWay/laravel-mix/tree/master/docs#readme)
+[To get started with Vite, please visit the documentation.](https://vitejs.dev/guide/)
 
 ```sh
-// Run all mix tasks...
+// Start the dev server...
 npm run dev
```
Comment (fiskhandlarn) [inline README.md:170]: `vite`

**Review by fiskhandlarn — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (vinkla) [reply README.md:173]: We've added these as npm scripts.

Comment (vinkla) [reply README.md:173]: No worries!

**Pull request merged**